### ISSUE TITLE
Support Habbo.swf for Rewritten versions

### ISSUE THREAD

**Issue by fp07** [ruffle-rs/ruffle#1339]
Support Habbo.swf https://gofile.io/d/y1WnXZ

The Habbo Rewritten scene is massive and still uses Flash Player, maybe one of the biggest community around if it comes to private servers on the browser:
- https://www.habblet.city/  (2000+ concurrent online users)
- https://www.habboon.pw/ (1000+ concurrent online users)
- https://www.habbocity.me/ (1000+ concurrent online users)
- https://peakrp.com/ (500+ concurrent online users)
- https://fresh-hotel.org/index (500+ concurrent online users)
- https://hangout.ph/ (500+ concurrent online users)
+ more than hundreds of websites with 100 - 500+ concurrent online users

A large proportion of teens playing these games would appreciate ruffle working for these versions. (may appeal to 100K+ players on a daily basis using this swf)

**Comment by ROBERT-MCDOWELL** [comment]
yes, they use also remote sharedobject and rtmfp..

**Comment by fp07** [comment]
What do you see for possibilities, for this swf on the short term before end of 2020? @ROBERT-MCDOWELL

**Comment by ROBERT-MCDOWELL** [comment]
Alhough I contacted twice the habbohotel developers without answer as they claim to have something like 25 millions users I think they have a plan to make habbo continue to run, maybe with cheerpX (they ask unreasonable $$$ / year), or for sure Ruffle can be a good option if some genius developer heroes can solve the Network/NetStrea/SharedObject classes. I think http3 will be a big opportunity to implement it on top of TCP and UDP.

**Comment by ROBERT-MCDOWELL** [comment]
I would see it around february/march 2021.

**Comment by fp07** [comment]
Indeed, cheerpX is around 2K for the license + an harman license which costs $25K, per year, per platform. Meaning you need $77K to support all platforms per year. Unreasonable indeed @ROBERT-MCDOWELL

**Comment by ROBERT-MCDOWELL** [comment]
there is also the amazing waflash https://github.com/vidkidz/waflash but no sources code for now.

**Comment by fp07** [comment]
I think the challenge for ruffle is the support for raw tcp connections. It needs I think always to use an bridge, until browsers support low tcp via websockets for example.

see https://github.com/ruffle-rs/ruffle/issues/905#issuecomment-707038785

**Comment by ROBERT-MCDOWELL** [comment]
http3 (QUIC) is raw socket, that's why I'm talking about it.

**Comment by fp07** [comment]
[Habbo.zip](https://github.com/ruffle-rs/ruffle/files/5425942/Habbo.zip)

**Comment by fp07** [comment]
Hopefully there is soon an update regarding this swf.

**Comment by ROBERT-MCDOWELL** [comment]
for ruffle dev team https://github.com/zenomt/rtmfp-cpp
